app/src/lib/source-map-support.ts

```diff
@@ -106,9 +106,20 @@ export function enableSourceMaps() {
  * perform the source mapping, it'll use the original error stack.
  */
 export function withSourceMappedStack(error: Error): Error {
+  // Guard against `throw "Foo"` which is totally a thing that can happen
+  if (typeof error === 'string') {
+    return {
+      name: 'StringError',
+      message: error,
+    }
+  }
+
   return {
     name: error.name,
-    message: error.message,
+    message:
+      typeof error.message === 'string'
+        ? error.message
+        : JSON.stringify(error.message),
```

Comment:
Should we use this instead?
```suggestion
        : JSON.stringify(error.message, Object.getOwnPropertyNames(error.message)),
```

From https://stackoverflow.com/a/26199752 as I wondered if there is any case where JSON.stringify wouldn't work as expected…